File: backend/models/train.py
from pydantic import BaseModel
from typing import Dict, Optional

class Train(BaseModel):
    id: str
    number: str
    name: str
    scheduled_arrival: str  # Format: "HH:MM"
    current_delay_mins: int
    avg_passengers: int
    class_breakdown: Dict[str, int]

class DelayEvent(BaseModel):
    train_id: str
    station_id: str
    platform_id: str
    delay_mins: int
    detected_at: str  # ISO Timestamp

File: frontend/src/types/train.ts
export interface Train {
  id: string;
  number: string;
  name: string;
  scheduled_arrival: string; // Format: "HH:MM"
  current_delay_mins: number;
  avg_passengers: number;
  class_breakdown: Record<string, number>;
}

export interface DelayEvent {
  train_id: string;
  station_id: string;
  platform_id: string;
  delay_mins: number;
  detected_at: string; // ISO timestamp
}
